### PR TITLE
[v1.22] ci: disable non-RBE cache for release branches

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

### CODE REVIEW

**Review by phlax — APPROVED** [review]
lgtm, thanks @lizan

File: .azure-pipelines/bazel.yml [hunk 1]
```diff
@@ -55,7 +55,6 @@ steps:
       GCP_SERVICE_ACCOUNT_KEY: $(GcpServiceAccountKey)
     ${{ if eq(parameters.rbe, false) }}:
       BAZEL_BUILD_EXTRA_OPTIONS: "${{ parameters.bazelBuildExtraOptions }}"
-      BAZEL_REMOTE_CACHE: $(LocalBuildCache)
```
Comment (alyssawilk) [inline .azure-pipelines/bazel.yml:58]: aren't we going to have to remember to do this for every branch we create?  Can we make this conditional on being in the main branch in yaml on main, so as we create new rease branches they do the right thing, and/or update "create a release" instructions so we remember to follow up if automating is too tricky?

Comment (phlax) [reply .azure-pipelines/bazel.yml:58]: i think we need a better solution - but for now its a case of just keeping the branches working - ill open a ticket ...

Comment (phlax) [reply .azure-pipelines/bazel.yml:58]: #24001

Comment (alyssawilk) [reply .azure-pipelines/bazel.yml:58]: Awesome, thanks

**Pull request merged**